devel-tools/compile-gromacs.sh

```diff
@@ -98,25 +103,22 @@ compile_gromacs_target() {
         ${CMAKE} --build "${GMX_BUILD_DIR}" --parallel $(nproc --all)
     ret_code=$?
 
-    if [ -n "${GROMACS_REGRESSION_TESTS}" ] ; then
+    if [ -n "${GROMACS_BUILD_TESTS}" ] ; then
         # Build the tests as well
         ${CMAKE} --build "${GMX_BUILD_DIR}" --target tests --parallel $(nproc --all)
         ret_code=$((${ret_code} || $?))
     fi
 
     if [ ${ret_code} = 0 ] ; then
-        pushd "${GMX_BUILD_DIR}"
-        ${CTEST} --output-on-failure
-        retcode=$?
-        if [ -n "${GMX_INSTALL_DIR}" ] ; then
-            if hash ninja >& /dev/null ; then
-                ninja install
-            else
-                make install
-            fi
-            ret_code=$((${ret_code} || $?))
+        if [ -n "${GROMACS_BUILD_TESTS}" ] && [ -z "${GITHUB_ACTION}" ] ; then
```

Comment:
This condition will never be true since GROMACS_BUILD_TESTS is set only when GITHUB_ACTION is set.
So, on Github Actions, this will just build the tests. 
I don't really understand the purpose.

Reply:
The purpose is to allow the CI jobs to build the tests (to catch potential errors), but not actually run them on GitHub Actions, because they are not supported by the GROMACS team for Linux. On top of that, we had numerous failures for standard GROMACS (`main` branch) on Windows and macOS.

Perhaps in the future we could run the GROMACS tests for stable release versions, but for `main` they generate too many false positives IMO.

Reply:
okay I understand.
but then, remove this condition no? Because i don't see the case where the condition is met and it's confusing imo.

Reply:
It would be met when we run the tests locally, no?

Reply:
oh yes indeed but only if the env variable GROMACS_REGRESSION_TESTS is set.
okay i see.